### ISSUE TITLE
CP Edit pages error if Matrix contains Number fieldtype - [v2.8.0.1 / PHP7.4]

### ISSUE THREAD

**Issue by michaelroper** [craftcms/cms#5353]
### Description
After the 2.8.0.1 update, I am getting errors when trying to edit entries that contain a Matrix field which in turn contains a 1st-party Number fieldtype. I tried adding a Number field directly to the field layout, and that _didn't_ seem to have the same problem, only if it is within a Matrix field.

The error is `Trying to access array offset on value of type int`, from the following line:

```php
// craft/app/etc/i18n/NumberFormatter.php(46)
34         {
35             if ($decimalFormat[$formatCounter] !== '#')
36             {
37                 break;
38             }
39         }
40 
41         $formatCounter += 1;
42 
43         // Find the starting decimal position in the value.
44         for ($valueCounter = strlen($value) - 1; $valueCounter >= 0; $valueCounter--)
45         {
46             if (!is_numeric($value[$valueCounter]))
47             {
48                 break;
49             }
50         }
51 
52         $valueCounter += 1;
53 
54         // Calculate how many decimals we're using.
55         $decimalLength = strlen($value) - $valueCounter;
56 
57         // Adjust the format for the number of decimals.
58         for ($finalCounter = $formatCounter; $finalCounter <= $decimalLength + $formatCounter; $finalCounter++)
```

### Steps to reproduce

1. Create a matrix field with a Number fieldtype, all default settings
2. Add that field to a entry layout. 
3. Try and create/edit an entry with that field layout.
4. see error above

### Additional info

- Craft version: v2.8.0.1
- PHP version: 7.4.0
- Database driver & version: MariaDB 10.4.10_1

**Comment by michaelroper** [comment]
Dropped back to PHP 7.3.12, and it works...

**Comment by brandonkelly** [comment]
Thanks for reporting that! Just released Craft 2.8.0.2 with a fix.
